contracts/ExecutionNodeVersionBeacon.cdc

```diff
@@ -0,0 +1,399 @@
+            } else if (self.major == other.major && self.minor > other.minor) {
+                return true
+            } else if (self.major == other.major && self.minor == other.minor && self.patch > other.patch) {
+                return true
+            } else {
+                return false
+            }
+        }
+
+        /// Returns true if Semver core is greater than or
+        /// equal to passed Semver core and false otherwise
+        pub fun coreGreaterThanOrEqualTo(_ other: Semver): Bool {
+            return self.coreGreaterThan(other) || self.coreEqualTo(other)
+        }
+
+        /// Returns true if Semver core is less than
+        /// passed Semver core and false otherwise
+        pub fun coreLessThan(_ other: Semver): Bool {
+            return !self.coreGreaterThanOrEqualTo(other)
+        }
+
+        /// Returns true if Semver core is less than or
+        /// equal to passed Semver core and false otherwise
+        pub fun coreLessThanOrEqualTo(_ other: Semver): Bool {
+            return self.coreLessThan(other) || self.coreEqualTo(other)
+        }
+
+        /// Returns true if Semver is equal to passed
+        /// Semver core and false otherwise
+        pub fun coreEqualTo(_ other: Semver): Bool {
+            return self.major == other.major && self.minor == other.minor && self.patch == other.patch
+        }
+
+        /// Returns true if Semver is *exactly* equal to passed
+        /// Semver and false otherwise
+        pub fun strictEqualTo(_ other: Semver): Bool {
+            return self.coreEqualTo(other) && self.preRelease == other.preRelease
+        }
+    }
+
+    /// Event emitted any time a change is made to versionTable
+    pub event ExecutionNodeVersionTableAddition(height: UInt64, version: Semver)
+    pub event ExecutionNodeVersionTableDeletion(height: UInt64, version: Semver)
+
+    /// Event emitted any time the version update buffer period or variance is updated
+    pub event ExecutionNodeVersionUpdateBufferChanged(newVersionUpdateBuffer: UInt64)
+    pub event ExecutionNodeVersionUpdateBufferVarianceChanged(newVersionUpdateBufferVariance: UFix64)
+
+    /// Canonical storage path for ExecutionNodeVersionAdmin
+    pub let ExecutionNodeVersionAdminStoragePath: StoragePath
+
+    /// Table dictating minimum version - {blockHeight: Semver}
+    /// Should be ordered by block height (index) which is
+    /// enforced by insertion/deletion
+    access(contract) let versionTable: {UInt64: Semver}
+
+    /// Number of blocks between block height when version boundary is defined and
+    /// the version boundary being defined. Nodes can expect that changes to the versionTable
+    /// they will have at least this long to respond to a version boundary
+    access(contract) var versionUpdateBuffer: UInt64
+
+    /// Set expectations regarding how much the versionUpdateBuffer can vary between version boundaries
+    access(contract) var versionUpdateBufferVariance: UFix64
```

Comment:
@m4ksio @janezpodhostnik Do you guys think it is needed for the first iteration of the contract?  Highly Opinionated here I think it is a little bit over-engineered because, as a resource owner, I can change the `versionUpdateBufferVariance` freely, and that will allow me to change the `versionUpdateBuffer` freely as well, so in the end, the sole purpose of having the `versionUpdateBufferVariance` got defeated.

If you think it is needed, we can leave this as it is. Otherwise, I am in favour of dropping it; if it is needed later, we can add it again.

Reply:
You make a good point Satyam. My thought behind introducing `versionUpdateBufferVariance` was to set expectations about how much the buffer would change between updates. 

To your point, it's probably overengineered. Especially because as you mentioned, the resource owner can change them as they wish without calling the designated methods or emitting the events. 

I think the best path forward for this iteration is to remove the variance and any references to it as well as set the `versionUpdateBuffer` to `access(self)`. That way changing the buffer value must go through the designated method and associated checks. The checks being that the change to the buffer doesn't cross a version boundary. 

Does that make sense?